### ISSUE TITLE
Rule for no newline after return statement not working

### ISSUE THREAD

**Issue by aaani** [eslint/eslint#8968]
**Tell us about your environment**

* eslint: 4.2.0
* node: 6.10.3
* npm: 3.10.10

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "node": true
    },
    "extends": ["eslint:recommended"],
    "parserOptions": {
        "ecmaVersion": 3,
        "sourceType": "module"
    },
    "rules" : {
        "padding-line-between-statements": [2,
            { "blankLine": "never", "prev": "return", "next": "*" }
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
function abc(){
    var x = 5;
    return x;

}

abc();
```

**What did you expect to happen?**
My goal here is to not allow any newlines after return and before the closing }. 

**What actually happened? Please include the actual, raw output from ESLint.**
When I run eslint on that code it doesn't give me error about the newline present just after the return statement.

**Comment by platinumazure** [comment]
You're using the `padding-line-between-statements` rule, which explicitly only covers newlines between statements. The end of your function block is not another statement.

This was by design, to avoid conflict with the [`padded-blocks`](http://eslint.org/docs/rules/padded-blocks) rule. You should configure and enable that rule to get the behavior you want.

Hope this helps!

**Comment by aaani** [comment]
Thanks @platinumazure, `padded-blocks` did the trick for me!
